### ISSUE TITLE
Apparent memory leak when checking a soldier's worldTransform property

### ISSUE THREAD

**Issue by MajorVictory** [VeniceUnleashed/VeniceUnleashed#605]
put this into any ext/client/\_\_init__.lua:
```lua
local testPlayer = nil
Events:Subscribe('UI:DrawHud', function()
    if (testPlayer == nil) then
        testPlayer = PlayerManager:GetLocalPlayer()
    end
    for i=1, 3000 do
        if (testPlayer ~= nil and testPlayer.soldier ~= nil and testPlayer.soldier.worldTransform ~= nil) then
            -- do literally nothing
        end
    end
end)
```

Once you spawn in, you'll get stuttering every few seconds, presumably due to garbage collection attempts, and eventually you'll run out of memory and crash.

**Comment by kapraran** [comment]
I also have some weird memory issues with 'UI:DrawHud'. A workaround that worked for me is to replace it with the `UpdatePass.UpdatePass_PreFrame` pass.

```lua
Events:Subscribe('UpdateManager:Update', function(deltaTime, updatePass)
  if updatePass ~= UpdatePass.UpdatePass_PreFrame then
    return
  end

  -- rest of the code...
end)
```

I'll upload a repro if you need more that what Major posted.

**Comment by kapraran** [comment]
[repro.zip](https://github.com/EmulatorNexus/VeniceUnleashed/files/6014768/repro.zip)

Steps to reproduce
1. Load the two included mods
2. Use the `XP2_Skybar Domination0 1` as map/mode
3. When ingame press `F1` to enable MKB
4. Toggle the `Show Text` checkbox many times and after some seconds the client crashes

I thought that my `vu-debug` mod may cause the issue but the **exact** same code that runs in the `UI:DrawHud` callback, runs without any issues if you put it in the event i posted above.

You can try it by yourself just by commenting the `UI:DrawHud` code and uncommenting the lines 60-73

**Comment by FlashHit** [comment]
I think the title should be more sth like:
`UI:DrawHud` is causing a memory leek

As this hasn' t really anything to do with the soldier's worldTransform.

**Comment by FlashHit** [comment]
> 
> 
> I also have some weird memory issues with 'UI:DrawHud'. A workaround that worked for me is to replace it with the `UpdatePass.UpdatePass_PreFrame` pass.
> 
> ```lua
> Events:Subscribe('UpdateManager:Update', function(deltaTime, updatePass)
>   if updatePass ~= UpdatePass.UpdatePass_PreFrame then
>     return
>   end
> 
>   -- rest of the code...
> end)
> ```
> 
> I'll upload a repro if you need more that what Major posted.

Just checked this, there is no difference for me. It has nothing to do with the event itself. 

------------------
Anyways I made a pretty simple repro that shows the issue:
```lua
Events:Subscribe('Engine:Update', function(p_DeltaTime, p_SimulationDeltaTime)
	for i = 1, 3000 do
		local s_Test = Vec4()
		-- local s_Test = 0
	end
end)
```
if you pass a float everything works fine, but as soon as you pass a frostbite type the memory is increasing until you crash.

Here again as a repro: 
[ReproMemoryLeak.zip](https://github.com/EmulatorNexus/VeniceUnleashed/files/6913984/ReproMemoryLeak.zip)

Someone should adjust the title.
Also this is no webui issue, the label is wrong I guess.

**Comment by FlashHit** [comment]
One last thing.
If you disable the `for loop` after you collected garbage and have like 2gb memory usage, it will still stay like that. So it seems like its not the case that the garbage collector just cant keep up, because otherwise it would collect it after I disabled the `for loop`, right? Seems that I will totally ignore that 'fb type' garbage and never collects it. And this doesn't seems right to me.
```lua
local s_Enable = true

Console:Register("disable", "Disable the memory leak", function(p_Args)
	s_Enable = false
end)

Events:Subscribe('Engine:Update', function(p_DeltaTime, p_SimulationDeltaTime)
	if s_Enable then
		for i = 1, 3000 do
			local s_Test = Vec3()
		end
	end
end)
```

In my test I collected in 1 minute 2gb memory, then disabled the `for loop`, and 20 minutes later it was still at 2gb memory. 


Doing this, works fine:
```lua
for i = 1, 3000 do
	local s_Test = {Vec3(), __mode = "kv"}
end
```
